docs/src/contributing/pull-request.md

```diff
@@ -0,0 +1,135 @@
+# The PR process for agave source code development
+
+## Preparation
+
+1. You’ll want a powerful machine for this. Building agave from sources takes a lot of RAM, make sure you have ~30GB to be sure it is enough.
+   1. Rust analyzer is quite capable of running on agave source tree, but expect high memory consumption
+2. Make a fork of `anza-xyz/agave` master branch under your own namespace in github. Let us assume it is called `ubercoder/agave`
+   1. Do not ever commit anything into the master branch of your fork. Keep it synced to upstream, this will make rebase process painless.
+   2. Whenever you want to pull agave code, pull it from your fork not from agave repo directly.
+3. Follow a guide to set up a reasonable rust development environment, e.g. [this one](https://github.com/anza-xyz/agave/?tab=readme-ov-file#building)
+   1. Make sure you can at least build the code before proceeding
+4. Plan the minimal viable set of changes you'd need to achieve your goals. Idea is to keep your PRs as small and simple as feasible.
```

Comment:
I think this topic of what should be the size of the PR should be addressed in more details. 

Consider scenario when a developer prepared a large chunk of changes to some component. Currently, it is implicitly assumed that he/she should split these changes into several PRs. Firstly, there might be different approaches to define the size of each PR. Secondly, having a chain of small PR might make a semantics of each PR to be obscure. Hence, there should be an issue which describes the context, describes a chain of changes and place of each PR in the whole chain of changes.

Reply:
I'm in full support of this. There are some decent ideas in CONTRIBUTING.md, which of them do you like?